src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js

```diff
@@ -0,0 +1,246 @@
+import { centeredSpinner, icon } from 'src/components/icons'
+import { textInput } from 'src/components/input'
+import { collapseStatus, failedIcon, runningIcon, statusIcon, successIcon } from 'src/components/job-common'
+import { FlexTable, Sortable, TextCell, TooltipCell } from 'src/components/table'
+import TooltipTrigger from 'src/components/TooltipTrigger'
+import { ajaxCaller } from 'src/libs/ajax'
+import { bucketBrowserUrl } from 'src/libs/auth'
+import colors from 'src/libs/colors'
+import { getConfig } from 'src/libs/config'
+import { reportError } from 'src/libs/error'
+import * as Nav from 'src/libs/nav'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer'
+
+
+const SubmissionDetails = _.flow(
+  forwardRef,
+  wrapWorkspace({
+    breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
+    title: 'Job History', activeTab: 'job history'
+  }),
+  ajaxCaller
+)((props, ref) => {
+  const { namespace, name, submissionId, ajax: { Workspaces }, workspace: { workspace: { bucketName } } } = props
+
+  /*
+   * State setup
+   */
+  const [submission, setSubmission] = useState({})
+  const [methodAccessible, setMethodAccessible] = useState()
+  const [statusFilter, setStatusFilter] = useState([])
+  const [textFilter, setTextFilter] = useState('')
+  const [sort, setSort] = useState({ field: 'workflowEntity', direction: 'asc' })
+
+  /*
+   * Data fetchers
+   */
+  const getSubmission = Workspaces.workspace(namespace, name).submission(submissionId).get
+  const initialize = async () => {
+    try {
+      const submission = await getSubmission()
+      setSubmission(submission)
+
+      try {
+        const { methodConfigurationName: configName, methodConfigurationNamespace: configNamespace } = submission
+        await Workspaces.workspace(namespace, name).methodConfig(configNamespace, configName).get()
+        setMethodAccessible(true)
+      } catch {
+        setMethodAccessible(false)
+      }
+    } catch (e) {
+      reportError('Unable to fetch submission details', e)
+    }
+  }
+
+  Utils.useOnMount(() => { initialize() })
+  useEffect(() => {
+    if (_.some(({ status }) => collapseStatus(status) === 'running', submission.workflows)) {
+      const timeout = setTimeout(async () => {
+        try {
+          const submission = await getSubmission()
+          setSubmission(submission)
+        } catch (e) {
+          reportError('Unable to update submission details', e)
+        }
+      }, 60000)
+      return () => clearTimeout(timeout)
+    }
+  }, [getSubmission, submission])
+
+  /*
+   * Sub-component constructors
+   */
+  const makeStatusLine = (iconFn, text) => div({ style: { display: 'flex', marginTop: '0.5rem', fontSize: 14 } }, [
+    iconFn({ marginRight: '0.5rem' }), text
+  ])
+
+  const makeSection = (label, children) => div({ style: { flexBasis: '33%', marginBottom: '1rem' } }, [
```

Comment:
These sections can sometimes get out of alignment at smaller screen sizes or if the content overflows. If the intent is to create a grid, they should probably be 'flex: none' with a defined width, and be able to deal with overflow somehow.